ui/src/app/core/job-manager.service.ts

```diff
@@ -103,6 +105,15 @@ export class JobManagerService {
       .catch((e) => this.handleError(e));
   }
 
+  updateJobLabels(id: string, req: UpdateJobLabelsRequest): Promise<UpdateJobLabelsResponse> {
+    return this.http.post(`${environment.apiUrl}/jobs/${id}/updateLabels`,
+      req,
+      new RequestOptions({headers: this.getHttpHeaders()}))
+      .toPromise()
+      .then((response) => response.status == 200)
```

Comment:
When you call then/catch on a promise, it returns a new promise with a potentially different resolved value. So this method returns a `Promise<boolean>`, which is probably not what you were aiming to do. You can just drop the `then`.